modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java

```diff
@@ -302,10 +312,11 @@ protected String getJDBCUrl(Map<String, ?> params) throws IOException {
         String host = (String) HOST.lookUp(params);
         String db = (String) DATABASE.lookUp(params);
         int port = (Integer) PORT.lookUp(params);
-        String url = "jdbc:postgresql" + "://" + host + ":" + port + "/" + db;
+        Boolean reWriteBatchedInserts = (Boolean) REWRITE_BATCHED_INSERTS.lookUp(params);
+        String url = "jdbc:postgresql" + "://" + host + ":" + port + "/" + db + "?reWriteBatchedInserts=" + reWriteBatchedInserts;
```

Comment:
Should there be a null check here like with the sslmode below? It would probably be better to make sure that the URL does not have `reWriteBatchedInserts=null` even if that doesn't cause an exception.

Reply:
i have updated the code to `boolean reWriteBatchedInserts = Boolean.TRUE.equals(REWRITE_BATCHED_INSERTS.lookUp(params));`

`Boolean.TRUE.equals(null)` will evaluate to `false`